eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/EnforceOverrideAnnotationCheck.java

```diff
@@ -0,0 +1,141 @@
+ */
+public class EnforceOverrideAnnotationCheck extends AbstractCheck {
+
+    /**
+     * Character separate package names in qualified name of java class.
+     */
+    private static final String PACKAGE_SEPARATOR = ".";
+
+    /**
+     * Full qualified name of the package.
+     */
+    private String packageName;
+
+    /**
+     * Full qualified name of the class.
+     */
+    private String className;
+
+    /**
+     * {@link Override Override} annotation name.
+     */
+    private static final String OVERRIDE = "Override";
+
+    /**
+     * Fully-qualified {@link Override Override} annotation name.
+     */
+    private static final String FQ_OVERRIDE = "java.lang." + OVERRIDE;
+
+    private static final String ANNOTATION_MISSING_OVERRIDE = "Must include @java.lang.Override or @Override annotation";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[]{TokenTypes.PACKAGE_DEF, TokenTypes.CLASS_DEF, TokenTypes.METHOD_DEF};
+    }
+
+    @Override
+    public void visitToken(DetailAST token) {
+        switch (token.getType()) {
+            case TokenTypes.PACKAGE_DEF:
+                packageName = extractQualifiedName(token.getFirstChild().getNextSibling());
+                break;
+            case TokenTypes.CLASS_DEF:
+                className = token.findFirstToken(TokenTypes.IDENT).getText();
+                break;
+            case TokenTypes.METHOD_DEF:
+                String methodName = token.findFirstToken(TokenTypes.IDENT).getText();
+
+                if (isMethodOverridden(methodName, getMethodParameters(token)) &&
+                    !AnnotationUtil.containsAnnotation(token, OVERRIDE) &&
+                    !AnnotationUtil.containsAnnotation(token, FQ_OVERRIDE)) {
+                    log(token, ANNOTATION_MISSING_OVERRIDE);
+                }
+            default:
+                // Checkstyle complains if there's no default block in switch
+                break;
+        }
+    }
+
+    /**
+     * Get name of class (with qualified package if specified) in {@code ast}.
+     *
+     * @param ast ast to extract class name from
+     * @return qualified name
+     */
+    private static String extractQualifiedName(DetailAST ast) {
+        return FullIdent.createFullIdent(ast).getText();
+    }
+
+    private Class<?>[] getMethodParameters(DetailAST token) {
+        DetailAST parameters = token.findFirstToken(TokenTypes.PARAMETERS);
+        DetailAST child = parameters.findFirstToken(TokenTypes.PARAMETER_DEF);
+        List<Class<?>> classes = new ArrayList<>();
+        while (child != null) {
+            if (child.getType() == TokenTypes.PARAMETER_DEF) {
+                classes.add(child.findFirstToken(TokenTypes.TYPE).getFirstChild().getClass());
```

Comment:
This is a little more intricate than calling `.getClass()`. `.getClass()` is getting the `Class` for the type being returned by `.getFirstChild()` and not the type of the parameter. This is also tricky with non-imported types such as `Integer` or `String`, where it may be easier to return `String[]` and match the class's method on parameter count, ordering, and parameter type names.